### ISSUE TITLE
AUR package for rust version

### ISSUE THREAD

**Issue by neitsab** [nspawn/nspawn#20]
Hey, following your comment in #19 I've created a PKGBUILD for the `rust_dev` branch which works perfectly, and indeed the new features are pretty awesome 🤩

I have two questions before publishing it:

1. how do you handle versions: you put `0.1.0` in the Cargo file, is that to be considered a release? So far I've created a `-git` package, meaning that it will track the branch HEAD. But if you do tags or releases, that's even better 😁

2. I've managed to generate the completion files for bash, fish and zsh by taking inspiration on another package so that they are automatically placed in the right directories after install, but for the life of me I couldn't find where to put the generated file for the Elvish shell. Do you have any idea if they use a standard directory like the others??

Here's the nearly finished PKGBUILD if you want to try it out:

```sh
# Maintainer: Bastien "neitsab" Traverse <neitsab@archlinux.org>

pkgname=nspawn-rust-git
pkgver=0.1.0
pkgrel=1
pkgdesc='A wrapper around machinectl for easy-deployment of nspawn.org containers'
url='https://github.com/nspawn/nspawn'
license=('GPL-3.0')
makedepends=('cargo' 'git')
provides=('nspawn')
conflicts=('nspawn')
arch=('i686' 'x86_64' 'armv6h' 'armv7h')
source=("$pkgname::git+${url}.git#branch=rust_dev")
sha256sums=('SKIP')

# pkgver() {
#   cd "${pkgname%-rust-git}"
#   printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short=7 HEAD)"
# }

prepare() {
  cd "$srcdir/$pkgname"
  export RUSTUP_TOOLCHAIN=stable
  cargo fetch --locked --target "$(rustc -vV | sed -n 's/host: //p')"
}

build() {
  cd "$srcdir/$pkgname"
  export RUSTUP_TOOLCHAIN=stable
  export CARGO_TARGET_DIR=target
  cargo build --frozen --release --all-features
}

check() {
  cd "$srcdir/$pkgname"
  export RUSTUP_TOOLCHAIN=stable
  cargo test --frozen --all-features
}

package() {
  cd "$srcdir/$pkgname"
  install -Dm0755 -t "$pkgdir/usr/bin/" "target/release/${pkgname%-rust-git}"

  # Shell completions
  target/release/${pkgname%-rust-git} --autocomplete=bash | install -Dm644 /dev/stdin "$pkgdir/usr/share/bash-completion/completions/nspawn"
  # No idea how to do it for elvish
  # target/release/${pkgname%-rust-git} --autocomplete=elvish | install -Dm644 /dev/stdin "$pkgdir/usr/share/..."
  target/release/${pkgname%-rust-git} --autocomplete=fish | install -Dm644 /dev/stdin "$pkgdir/usr/share/fish/vendor_completions.d/nspawn.fish"
  target/release/${pkgname%-rust-git} --autocomplete=zsh | install -Dm644 /dev/stdin "$pkgdir/usr/share/zsh/site-functions/_nspawn"
}
```

PS: btw there is no license file in this branch, so I just went with the original GPLv3

Cheers!

**Comment by Edu4rdSHL** [comment]
Hi, thank you for your efforts! The PKGBUILD looks good to me, just a few notes:

- Leave `pkgver()` uncommented, it's a -git package and should take the version from there.
- Ignore elvish shell for now.

Additionally, I have emailed the maintainer of the AUR `nspawn` package and I hope that we can bring the rust version there soon (he already replied kindly and said that there aren't problem with that).

Cheers, and thanks for testing! Please check out https://github.com/nspawn/nspawn/discussions/21 for updates, I will be tracking features and more there.

**Comment by neitsab** [comment]
> * Leave `pkgver()` uncommented, it's a -git package and should take the version from there.

Yeah I had prepared it but commented it out in case you wanted to go with a regular pkgver (in that case I would have converted it to a non-VCS package).

> Additionally, I have emailed the maintainer of the AUR `nspawn` package and I hope that we can bring the rust version there soon (he already replied kindly and said that there aren't problem with that).

This is great to hear! Actually that's what I was hoping for, the rust branch becoming the main one and the existing AUR package integrating those changes.

(Though if that's the case I'll have to rename this package to simply `nspawn-git` instead of `nspawn-rust-git`, but there are worst things in life 😁)

In any case, the package is now live at https://aur.archlinux.org/packages/nspawn-rust-git!

**Comment by Edu4rdSHL** [comment]
I already have maintainer permissions on both `nspawn` and `nspawn-git`! If you like, we can include your `PKGBUILD` into `nspawn-git`.

**Comment by neitsab** [comment]
Hey, sorry for the delay! If you're fine with it I'm more than happy for my PKGBUILD to be used for `nspawn-git`. Let me know if you go ahead with this, so that I can delete `nspawn-rust-git` and avoid duplication :-)

**Comment by Edu4rdSHL** [comment]
I will look into it, thanks!

**Comment by Edu4rdSHL** [comment]
Hi, the main packages `nspawn` and `nspawn-git` are now on the Rust branch (which is on main now).
